doc/architecture.md

```diff
@@ -0,0 +1,96 @@
+# Architecture
+
+The architecture of the system tests consist out of two layers.
+The interface between the layers are docker images.
+
+## Grammar
+
+```
+// terminals
+System         = string
+TestClass      = string
+Feature        = string
+Dockerfile     = "Dockerfile"
+TestPrefix     = "Test"
+.              = "."
+_              = "_"
+
+// Non-terminals
+SystemSpec     = System [. Feature]*
```

Comment:
So a `SystemSpec` can only have one feature. `Arch.Boost160` is ok, `Arch.Boost160.PETSc3.8` is not`. Do I understand that correctly?

Reply:
The wildcard means 0 or more features